python/paddle/fluid/tests/unittests/dygraph_to_static/test_lac.py

```diff
@@ -0,0 +1,570 @@
+# Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+import math
+import time
+import numpy as np
+import unittest
+
+import paddle.fluid as fluid
+from paddle.fluid.dygraph import to_variable
+from paddle.fluid.dygraph import Embedding, Linear, GRUUnit
+from paddle.fluid.dygraph import declarative, ProgramTranslator
+
+SEED = 2020
+program_translator = ProgramTranslator()
+
+
+class DynamicGRU(fluid.dygraph.Layer):
+    def __init__(self,
+                 size,
+                 h_0=None,
+                 param_attr=None,
+                 bias_attr=None,
+                 is_reverse=False,
+                 gate_activation='sigmoid',
+                 candidate_activation='tanh',
+                 origin_mode=False,
+                 init_size=None):
+        super(DynamicGRU, self).__init__()
+
+        self.gru_unit = GRUUnit(
+            size * 3,
+            param_attr=param_attr,
+            bias_attr=bias_attr,
+            activation=candidate_activation,
+            gate_activation=gate_activation,
+            origin_mode=origin_mode)
+
+        self.size = size
+        self.h_0 = h_0
+        self.is_reverse = is_reverse
+
+    def forward(self, inputs):
+        inputs = to_variable(inputs)
+        hidden = to_variable(self.h_0)
+        hidden.stop_gradient = True
```

Comment:
why call `to_variable` here?  In the original model, `to_variable` is not called.
And why set stop_gradient = True?

Reply:
1. why call `to_variable`?
Because in the following code, `hidden` will be assigned a new shape [-1, 128] but it has been modified into `Parameter` in Dy2stat. We don't allow parameter hold shape with -1 which will raise a error if you call `program.clone()`.
Here  I call `to_variable` to make a new var with same data of `self.h_0` to avoid the above problem. Maybe we should give a warning to notice users don't use in the way.

2. why set stop_gradient = True?
Because hidden is modified to Parameter with trainable=True and shall be not updated, so it don't hold gradient data.